routes/userRoutes.js

```diff
@@ -12,34 +12,41 @@ const createToken = require('../utils/createToken');
  * @access  Private
  */
 router.put('/:id', passport.authenticate('jwt', { session: false }), (req, res) =>
-	User.findById(req.params.id)
-		.then(user => {
-			if (!user) {
-				return res.status(404).json({ message: 'User not found' });
-			}
-			if (user._id.toHexString() !== req.user._id.toHexString()) {
-				return res.status(401).json({ message: 'Unauthorized' });
-			}
-			// TODO: Run validation for name
-			// image URL validation
-			if (req.body.imgUrl) {
-				const validUrl = validator.isURL(req.body.imgUrl, {
-					protocols: ['http', 'https'],
-					require_protocol: true,
-					require_tld: true,
-				});
-				if (!validUrl)
-					return res.status(400).json({ imgUrl: 'Please enter a valid image URL' });
+	User.findById(req.params.id).then(user => {
+		if (!user) {
+			return res.status(404).json({ message: 'User not found' });
+		}
+		if (user._id.toHexString() !== req.user._id.toHexString()) {
+			return res.status(401).json({ message: 'Unauthorized' });
+		}
+		// DT TODO: Discuss how to handle no values provided on update request
+		// Should we just return existent user object or throw err?
```

Comment:
I vote just returning the existent user object. If the user is updating their profile, I think it seems a little aggressive to throw an error just because they didn't change anything. The message alongside the response can still signify that nothing was actually updated.

Reply:
That would be the best approach indeed. Tnx!